src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -169,38 +189,22 @@ CHIP_ERROR WiFiManager::Connect(const ByteSpan & ssid, const ByteSpan & credenti
 
     mHandling.mOnConnectionSuccess = handling.mOnConnectionSuccess;
     mHandling.mOnConnectionFailed  = handling.mOnConnectionFailed;
+    mHandling.mConnectionTimeoutMs = handling.mConnectionTimeoutMs;
 
     mWiFiState = WIFI_STATE_ASSOCIATING;
 
-    struct net_if * iface = InetUtils::GetInterface();
-
-    VerifyOrReturnError(nullptr != iface, CHIP_ERROR_INTERNAL);
-
-    // We can use pointers for ssid and credentials without clone them because setting new ones is blocked during associating.
-    mWiFiParams.ssid        = ssid.data();
-    mWiFiParams.ssid_length = ssid.size();
-    mWiFiParams.psk         = const_cast<unsigned char *>(credentials.data());
-    mWiFiParams.psk_length  = credentials.size();
-    mWiFiParams.security    = WIFI_SECURITY_TYPE_PSK;
-    mWiFiParams.timeout     = handling.mConnectionTimeoutMs.count();
-    mWiFiParams.channel     = WIFI_CHANNEL_ANY;
-
-    if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface, &mWiFiParams, sizeof(ConnectionParams)))
-    {
-        ChipLogError(DeviceLayer, "Connection request failed");
-        if (Instance().mHandling.mOnConnectionFailed)
-            Instance().mHandling.mOnConnectionFailed();
-        return CHIP_ERROR_INTERNAL;
-    }
-
-    ChipLogError(DeviceLayer, "Connection to %s requested", ssid.data());
-
+    // Store SSID and credentials and perform the scan to detect the security mode supported by the AP.
+    // Connect() will be called in the callback when we have the SSID match.
+    mCurrentCredentials = credentials;
```

Comment:
Who is responsible for allocating/deallocating memory referenced by these spans?

Reply:
Hmm good question. As far as I understand the ByteSpan is just a wrapper for a raw pointer and data length and it does not do any deep copy under the hood. So essentially these spans (a.k.a underlying buffers) are owned by the `NrfWiFiDriver' instance (staging network object). Sounds like sth error prone. Maybe better have a local copy.

Reply:
Yes, I think we should copy them or document the ownership explicitly.